packages/order-utils/src/order_validation_utils.ts

```diff
@@ -106,6 +108,51 @@ export class OrderValidationUtils {
             TransferType.Fee,
         );
     }
+    // TODO(xianny): remove this method once the smart contracts have been refactored
+    // to return helpful revert reasons instead of ORDER_UNFILLABLE. Instruct devs
+    // to make "calls" to validate transfers
+    /**
+     * Validate the transfer from the maker to the taker. This is simulated on-chain
+     * via an eth_call. If this call fails, the asset is currently nontransferable.
+     * @param exchangeTradeEmulator ExchangeTradeEmulator to use
+     * @param signedOrder SignedOrder of interest
+     * @param makerAssetAmount Amount to transfer from the maker
+     * @param takerAddress The address to transfer to, defaults to signedOrder.takerAddress
+     */
+    public static async validateMakerTransferThrowIfInvalidAsync(
+        exchangeTradeEmulator: ExchangeTransferSimulator,
+        exchangeContract: ExchangeContract,
```

Comment:
Let's not pass the `exchangeContract` in here. That's a bit "heavy-duty". We have the `assetProxy` addresses for each networkId in `@0x/contract-addresses`, let's get them from there instead.

Reply:
yeah, I would love to get rid of `exchangeContract`! Do you think we should pass the `networkId` as an argument here? I'm more inclined to passing the `assetProxyAddress` as an arg, since the only use for `networkId` here is to fetch it. Unless there's a better way of getting it in scope?

Reply:
If we ask them to pass in the `assetProxyAddress`, then they need to decode the `makerAssetData` themselves and figure out which `assetProxyAddress` to pass in. I think `networkId` is more seamless for the developer. We do the magic under-the-hood.